code/_onclick/hud/follow_alert.dm

```diff
@@ -0,0 +1,157 @@
+/obj/screen/alert/augury/proc/on_following_qdel(atom/movable/A)
+	SIGNAL_HANDLER  // COMSIG_PARENT_QDELETING
+	for(var/atom/movable/follower in followers)
+		follower.stop_orbit()
+
+	follow_target = get_next_target()
+	if(follow_target)
+		if(time_between_switches > 0)
+			addtimer(CALLBACK(src, PROC_REF(change_targets), follow_target), time_between_switches)  // add a second so people can see what exploded
+		else
+			change_targets(follow_target)
+
+/**
+ * Change the atom that everyone is currently following, moving everyone to the new object.
+ *
+ * Arguments:
+ * * next_to - The next atom to follow.
+ */
+/obj/screen/alert/augury/proc/change_targets(atom/movable/next_to)
+	// unregister first so we aren't bombarded when changing orbits
+	if(isnull(next_to))
+		return
+	if(follow_target)
+		UnregisterSignal(follow_target, COMSIG_ATOM_ORBIT_STOP)
+		UnregisterSignal(follow_target, COMSIG_PARENT_QDELETING)
+	follow_target = next_to
+	RegisterSignal(follow_target, COMSIG_PARENT_QDELETING, PROC_REF(on_following_qdel), override = TRUE)  // override our qdeleting signal
+	for(var/atom/movable/M in followers)
+		M.orbit(follow_target)
+
+	RegisterSignal(follow_target, COMSIG_ATOM_ORBIT_STOP, PROC_REF(remove_follower_on_stop_orbit))
+
+/obj/screen/alert/augury/proc/add_follower(atom/movable/follower)
+	followers |= follower
+	follower.orbit(follow_target)
+
+/obj/screen/alert/augury/proc/remove_follower(atom/movable/follower)
+	followers -= follower
+	follower.stop_orbit()
+
+/obj/screen/alert/augury/proc/get_next_target()
+	if(!length(next_targets))
+		return null
+
+	var/atom/movable/target
+
+	while(QDELETED(target) && length(next_targets))
+		target = next_targets[length(next_targets)]  // pop the (hopefully most recent)
+		next_targets.Remove(target)
+
+	return target
+
+/// Called when someone stops orbiting our followed object, so they can actually get out of the loop.
+/obj/screen/alert/augury/proc/remove_follower_on_stop_orbit(atom/movable/followed, atom/movable/follower)
+	SIGNAL_HANDLER  // COMSIG_ATOM_ORBIT_STOP
+	if(locateUID(follower.orbiting_uid) != follow_target)
+		remove_follower(follower)  // don't try to stop the orbit again
+
+/// Meteor alert.
+/// Appears during a meteor storm and allows for auto-following of debris.
+/obj/screen/alert/augury/meteor
+	name = "Meteors incoming!"
+	desc = "Click to automatically follow debris, and click again to stop."
+
+/obj/screen/alert/augury/meteor/Initialize(mapload)
+	var/image/meteor_img = image(icon = 'icons/obj/meteor.dmi', icon_state = "flaming")
+	. = ..(mapload, alert_overlay_override = meteor_img)
+	START_PROCESSING(SSfastprocess, src)
+
+/obj/screen/alert/augury/meteor/Destroy(force)
+	. = ..()
+	STOP_PROCESSING(SSfastprocess, src)
+
+/obj/screen/alert/augury/meteor/process()
+	var/overridden = FALSE
+	for(var/obj/effect/meteor/M in GLOB.meteor_list)
+		if(!is_station_level(M.z))
+			continue  // don't worry about endlessly looping
+		if(istype(M, /obj/effect/meteor/tunguska))
```

Comment:
will there ever be two Tunguskas at once?

Reply:
Unlikely, but probably possible

Reply:
would you mind testing it to see if there is a rubberbanding effect?

Reply:
Just tested, seems like it's pretty good at following just a single tunguska. It looks like it may jump to a new one occasionally, but rarely, and is generally good enough at just following a single target.
Did notice some weird re-orbiting things happening though, which should be fixed in the next commit.